### PR TITLE
State push returns error if not authenticated

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-727" title="DX-727" target="_blank">StoryDX-727</a>  `state push` tells user to run `state auth` if not authenticated
  </td></table>
  <br />

### CODE REVIEW

**Comment by MDrakos** [comment]
The integration tests are that are failing is likely because of [this](https://activestatef.atlassian.net/browse/DX-713),

File: internal/testhelpers/e2e/session.go [hunk 1]
```diff
@@ -324,7 +324,7 @@ func (s *Session) LoginUser(userName string) {
 // LoginAsPersistentUser is a common test case after which an integration test user should be logged in to the platform
 func (s *Session) LoginAsPersistentUser() {
 	p := s.SpawnWithOpts(
-		WithArgs("auth", "--username", PersistentUsername, "--password", PersistentPassword),
+		WithArgs("auth", "--interactive", "--username", PersistentUsername, "--password", PersistentPassword),
```
Comment (mitchell-as) [inline internal/testhelpers/e2e/session.go:327]: This is actually not needed. Having `--username` automatically turns on `--interactive`.

**Review by mitchell-as — CHANGES_REQUESTED** [review]
One small change. Other than that, looks great!

**Review by mitchell-as — DISMISSED** [review]

File: internal/runners/push/push.go [hunk 2]
```diff
@@ -248,11 +246,7 @@ func (r *Push) Run(params PushParams) error {
 
 func (r *Push) verifyInput() error {
 	if !r.auth.Authenticated() {
-		err := authlet.RequireAuthentication(locale.Tl("auth_required_push", "You need to be authenticated to push a local project to the ActiveState Platform"), r.config, r.out, r.prompt, r.auth)
-		if err != nil {
-			return locale.WrapInputError(err, "err_push_auth", "Failed to authenticate")
-		}
-		r.out.Notice("") // Add line break to ensure output doesn't stick together
+		return locale.NewInputError("err_push_not_authenticated", "In order to update your project you need to be authenticated, please run '[ACTIONABLE]state auth[/RESET] to authenticate.")
```
Comment (Naatan) [inline internal/runners/push/push.go:249]:
```suggestion
		return locale.NewInputError("err_push_not_authenticated", "In order to update your project you need to be authenticated, please run '[ACTIONABLE]state auth[/RESET]' to authenticate.")
```

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**